src/python_testing/TC_MOD_2_3.py

```diff
@@ -0,0 +1,260 @@
+
+        found_var = False
+        for EFS in result.extensionFieldSetStructs:
+            if EFS.clusterID != Clusters.Objects.ModeSelect.id:
+                continue
+
+            for AV in EFS.attributeValueList:
+                if AV.attributeID == current_mode_attribute_id:
+                    asserts.assert_equal(AV.valueUnsigned8, mode_select_initial_mode, "View Scene failed on CurrentMode")
+                    found_var = True
+        asserts.assert_true(found_var, "View Scene failed to find CurrentMode attribute in ExtensionFieldSets")
+
+        self.step("5a")
+        result = await self.TH1.SendCommand(
+            self.dut_node_id, self.matter_test_config.endpoint,
+            Clusters.ScenesManagement.Commands.AddScene(
+                self.kGroup1,
+                0x02,
+                0,
+                "ModeSelect Scene",
+                [
+                    Clusters.ScenesManagement.Structs.ExtensionFieldSetStruct(
+                        clusterID=Clusters.Objects.ModeSelect.id,
+                        attributeValueList=[
+                            Clusters.ScenesManagement.Structs.AttributeValuePairStruct(
+                                attributeID=current_mode_attribute_id, valueUnsigned8=mode_select_m1_mode)
+                        ]
+                    )
+                ]
+
+            )
+        )
+        asserts.assert_equal(result.status, Status.Success, "Add Scene failed on status")
+        asserts.assert_equal(result.groupID, self.kGroup1, "Add Scene failed on groupID")
+        asserts.assert_equal(result.sceneID, 0x02, "Add Scene failed on sceneID")
+
+        self.step("5b")
+        result = await self.TH1.SendCommand(self.dut_node_id, self.matter_test_config.endpoint, Clusters.ScenesManagement.Commands.ViewScene(self.kGroup1, 0x02))
+        asserts.assert_equal(result.status, Status.Success, "View Scene failed on status")
+        asserts.assert_equal(result.groupID, self.kGroup1, "View Scene failed on groupID")
+        asserts.assert_equal(result.sceneID, 0x02, "View Scene failed on sceneID")
+        asserts.assert_equal(result.transitionTime, 0, "View Scene failed on transitionTime")
+
+        found_var = False
+
+        for EFS in result.extensionFieldSetStructs:
+            if EFS.clusterID != Clusters.Objects.ModeSelect.id:
+                continue
+
+            for AV in EFS.attributeValueList:
+                if AV.attributeID == current_mode_attribute_id:
+                    asserts.assert_equal(AV.valueUnsigned8, mode_select_m1_mode,
+                                         "View Scene failed on CurrentMode value")
+                    found_var = True
+        asserts.assert_true(found_var, "View Scene failed to find CurrentMode attribute in ExtensionFieldSets")
+
+        self.step("6a")
+        await self.TH1.SendCommand(self.dut_node_id, self.matter_test_config.endpoint, Clusters.ScenesManagement.Commands.RecallScene(self.kGroup1, 0x02, 0))
```

Comment:
Not sure if it makes any difference but the test plan says to `omit` the transition time and you do that on step 7a but not here...

```suggestion
        await self.TH1.SendCommand(self.dut_node_id, self.matter_test_config.endpoint, Clusters.ScenesManagement.Commands.RecallScene(self.kGroup1, 0x02))
```